lib/libmbedtls/core/tee_lmd_provider.c

```diff
@@ -0,0 +1,586 @@
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+
+TEE_Result crypto_aes_ccm_update_payload(void *ctx __unused,
+					 TEE_OperationMode mode __unused,
+					 const uint8_t *src_data __unused,
+					 size_t len __unused,
+					 uint8_t *dst_data __unused)
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+
+TEE_Result crypto_aes_ccm_enc_final(void *ctx __unused,
+				    const uint8_t *src_data __unused,
+				    size_t len __unused,
+				    uint8_t *dst_data __unused,
+				    uint8_t *dst_tag __unused,
+				    size_t *dst_tag_len __unused)
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+
+TEE_Result crypto_aes_ccm_dec_final(void *ctx __unused,
+				    const uint8_t *src_data __unused,
+				    size_t len __unused,
+				    uint8_t *dst_data __unused,
+				    const uint8_t *tag __unused,
+				    size_t tag_len __unused)
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+
+void crypto_aes_ccm_final(void *ctx __unused)
+{
+}
+#endif /*CFG_CRYPTO_CCM*/
+
+#if defined(CFG_CRYPTO_AES_GCM_FROM_CRYPTOLIB)
+TEE_Result crypto_aes_gcm_alloc_ctx(void **ctx_ret __unused)
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+
+void crypto_aes_gcm_free_ctx(void *ctx __unused)
+{
+	if (ctx)
+		assert(0);
+}
+
+void crypto_aes_gcm_copy_state(void *dst_ctx __unused, void *src_ctx __unused)
+{
+	assert(0);
+}
+
+TEE_Result crypto_aes_gcm_init(void *ctx __unused,
+			       TEE_OperationMode mode __unused,
+			       const uint8_t *key __unused,
+			       size_t key_len __unused,
+			       const uint8_t *nonce __unused,
+			       size_t nonce_len __unused,
+			       size_t tag_len __unused)
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+
+TEE_Result crypto_aes_gcm_update_aad(void *ctx __unused,
+				const uint8_t *data __unused,
+				size_t len __unused)
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+
+TEE_Result crypto_aes_gcm_update_payload(void *ctx __unused,
+					 TEE_OperationMode mode __unused,
+					 const uint8_t *src_data __unused,
+					 size_t len __unused,
+					 uint8_t *dst_data __unused)
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+
+TEE_Result crypto_aes_gcm_enc_final(void *ctx __unused,
+				    const uint8_t *src_data __unused,
+				    size_t len __unused,
+				    uint8_t *dst_data __unused,
+				    uint8_t *dst_tag __unused,
+				    size_t *dst_tag_len __unused)
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+
+TEE_Result crypto_aes_gcm_dec_final(void *ctx __unused,
+				    const uint8_t *src_data __unused,
+				    size_t len __unused,
+				    uint8_t *dst_data __unused,
+				    const uint8_t *tag __unused,
+				    size_t tag_len __unused)
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+
+void crypto_aes_gcm_final(void *ctx __unused)
+{
+	if (ctx)
+		assert(0);
+}
+#endif /*CFG_CRYPTO_AES_GCM_FROM_CRYPTOLIB*/
+
+/******************************************************************************
+ * Pseudo Random Number Generator
+ ******************************************************************************/
+TEE_Result __weak crypto_rng_read(void *buf __unused, size_t blen __unused)
```

Comment:
This function doesn't belong ere any more.

Reply:
Yes, I will remove PRNG patch.